keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1235 @@
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerPolicyOptions`
+  - Components depending on the feature gate: `kubelet`
+- [x] Other
+  - Describe the mechanism: Change the `kubelet` configuration to set a `CPUManager` policy of static then setting the policy option of `align-cpus-by-uncorecache`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? Yes, a `kubelet` restart is required for changes to take place.
+
+###### Does enabling the feature change any default behavior?
+
+No, to enable this feature, it must be explicitly set in the `CPUManager` static policy and the policy option `align-cpus-by-uncorecache` must be set.
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled it will just require a restart of `kubelet`. The Kubelet configuration will need to be set with the static policy option and align-cpus-by-uncorecache flag removed.
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Feature will be enabled. Proper drain of node and restart of kubelet required. Feature is not intended to be enabled/disabled dynamically, similar to static policy.
+
+###### Are there any tests for feature enablement/disablement?
+
+No, not intended to be enabled/disabled dynamically. 
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+### Rollout, Upgrade and Rollback Planning
```

Comment:
This section is not required for alpha but I encourage thinking through these questions.

@ffromani could you help with metrics or items that we would want for operation?

Should we introduce metrics on nodes that may have this cache?

Reply:
this is an in interesting ask but seems again a more general concern.
I think it would be beneficial to have metrics to track usage of full L3 core groups and total available L3 core groups. Same could possibly go for full physical cpus and full NUMA nodes maybe. I'm thinking if this can be served better by a separate PR, and/or if this is a too low-level metric.

Reply:
Wouldn't we just let cadvisor report this back? That's where we get the information

Reply:
Idk. This section is about thinking through these things. If those metrics exist in Prometheus with cadvisor that could be fine but we’d want to call out that we would use those.

I am also thinking through how does one know if this feature is working via metrics or the feature is causing an issue. I’m not very familiar with CPUManagwr so maybe this is not applicable. But it’s worth discussing.

As @ffromani knows, neglecting this metrics bites us in the butt down the road.

Reply:
Lack of feedback about the feature working or not is a concern and it was highlighted already multiple times during reviews (especially PRR), so it is totally something to think about. I'm still thinking about how to expose metrics/feedback in some form, I'll elaborate ASAP

Reply:
here's my proposal: https://github.com/kubernetes/kubernetes/pull/127155 we can either move forward with that PR or you can pull my commit in the implementation PR. If we like this direction, we can report alignment by LLC group, which gives feedback about this feature working or not.

Reply:
Okay lets take your commit and pull into the PR